### ISSUE TITLE
Indentation error causes pretext stage to train improperly

### ISSUE THREAD

**Issue by cosmos041389** [zamanzadeh/CARLA#13]
Hello,
First of all, I sincerely appreciate your progressive research and its approaches.

However, I found an indentation error that prevents the pretext stage from training properly when I executed the example code.
More precisely, it only iterates over a single batch per epoch.
I discovered that the issue is caused by an indentation error in `utils/train_utils.py:44`.
```
def pretext_train(train_loader, model, criterion, optimizer, epoch, prev_loss):

    losses = AverageMeter('Loss', ':.4e')
    progress = ProgressMeter(len(train_loader),
        [losses],
        prefix="Epoch: [{}]".format(epoch+1))

    model.train()

    for i, batch in enumerate(train_loader):
        ts_org = batch['ts_org']
        ts_w_augmented = batch['ts_w_augment']
        ts_ss_augmented = batch['ts_ss_augment']

        if ts_org.ndim == 3:
            b, w, h = ts_org.shape
        else:
            b, w = ts_org.shape
            h =1

        input_: Tensor = torch.cat([torch.from_numpy(ts_org).float(), torch.from_numpy(ts_w_augmented).float(), torch.from_numpy(ts_ss_augmented).float()], dim=0)
        input_ = input_.view(b*3, h, w)

        output = model(input_)
        # output = output.view(b, 3, -1)
        loss = criterion(output, current_loss = prev_loss)
        losses.update(loss.item())
        prev_loss = losses

        optimizer.zero_grad()
        loss.backward()
        optimizer.step()

        if i % 10 == 0:
            progress.display(i)

        return loss # <-- This should be out of the loop
```
The `return loss` statement should be moved one indentation level up so that it does not exit the loop prematurely.
Additionally, in `losses/losses.py:116`, you should change `current_loss` to `current_loss.val` because `current_loss` is an instance of `AverageMeter`. 
This ensures that the loss calculation uses the actual loss value.
Thank you for taking the time to read my issue.
Best regards,

**Comment by zamanzadeh** [comment]
Hello,

Thank you for your kind words and for bringing this issue to my attention. I've revised the code after the recent update, and the indentation error and loss value have been corrected. Please pull the latest changes and let me know if it resolves the problem for you.

Thank you once again for your contribution, and please do not hesitate to reach out if there are any other issues or if you have further suggestions.

Best
